src/components/shared/InputFields/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import classnames from 'classnames';
+import VisibilityOff from '../Icons/visibility_off';
+import Visibility from '../Icons/visibility';
+
+export enum InputVariants {
+  Simple = 'simple',
+  Error = 'error',
+  Password = 'password',
+}
+
+interface InputProps {
+  variant: string;
+  text: string;
+  placeholder: string;
+  required: boolean;
+  disabled?: boolean;
+}
+
+const InputField = ({ variant, text, required, placeholder, disabled }: InputProps) => {
+  const [passwordShown, setPasswordShown] = useState(false);
```

Comment:
Completely personal preference, WDYT? With the prefix `is` it clear that you are talking about a boolean
```suggestion
  const [isPasswordShown, setIsPasswordShown] = useState(false);
```